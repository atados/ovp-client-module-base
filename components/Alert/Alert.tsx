import React, { useState } from 'react'
import Icon from '~/components/Icon'

interface AlertProps {
  children: React.ReactNode
  bgClassName?: string
  onClose?: Function
}

const Alert: React.FC<AlertProps> = ({ children, bgClassName, onClose }) => {
  const [open, setOpen] = useState<boolean>(true)
  const handleClick = () => {
    if (onClose) onClose()
    setOpen(false)
  }

  if (!open) {
    return null
  }

  return (
    <div
      className={`w-full min-h-4 flex justify-center items-center py-2 px-4 bg-orange-400 ${bgClassName}`}
    >
      {children}
      <button
        role="close"
        onClick={handleClick}
        className="absolute right-0 mr-2 cursor-pointer"
      >
        <Icon name="close" />
      </button>
    </div>
  )
}

Alert.defaultProps = {
  bgClassName: 'bg-orange-400',
}

export default Alert
