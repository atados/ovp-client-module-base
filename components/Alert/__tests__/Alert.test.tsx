import React from 'react'
import renderer from 'react-test-renderer'

import Alert from '..'
import { render, fireEvent } from '@testing-library/react'

describe('Alert', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Alert>foo</Alert>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should call onClose prop', () => {
    const testClose = jest.fn()
    const root = render(<Alert onClose={testClose}>foo</Alert>)

    fireEvent.click(root.getByRole('close'))

    expect(testClose).toHaveBeenCalled()
  })
})
