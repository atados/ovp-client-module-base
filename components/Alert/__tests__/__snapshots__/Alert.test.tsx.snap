// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders correctly 1`] = `
<div
  className="w-full min-h-4 flex justify-center items-center py-2 px-4 bg-orange-400 bg-orange-400"
>
  foo
  <button
    className="absolute right-0 mr-2 cursor-pointer"
    onClick={[Function]}
    role="close"
  >
    <span
      className="icon im"
    >
      close
    </span>
  </button>
</div>
`;
